### PR TITLE
fix functional tests to work with mint

### PR DESCRIPTION
**Pull request by poornas**
- Added copy objects with conditions, remove object and stat custom content-type to existing tests
- use data files from /mint/data

### CODE REVIEW

File: src/test/functional/functional-tests.js [hunk 1]
```diff
@@ -337,6 +389,7 @@ describe('functional tests', function() {
   })
 
   describe('bucket policy', () => {
+    "use strict;"
```
Comment (krishnasrinivas) [inline src/test/functional/functional-tests.js]: @poornas how come you have put this here?

File: src/test/functional/functional-tests.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /*
  * Minio Javascript Library for Amazon S3 Compatible Cloud Storage, (C) 2015 Minio, Inc.
- *
+ a*
```
Comment (harshavardhana) [inline src/test/functional/functional-tests.js]: ??

File: src/test/functional/functional-tests.js [hunk 3]
```diff
@@ -14,75 +14,74 @@
...
+const os = require('os');
+const stream = require('stream');
+const crypto = require('crypto');
+const async = require('async');
+const _ = require('lodash');
+const fs = require('fs');
+const http = require('http');
+const https = require('https');
+const url = require('url');
+const chai = require('chai');
```
Comment (krishnasrinivas) [inline src/test/functional/functional-tests.js]:
```
import { assert } from 'chai'
```

this is an often used pattern when you need to use one or two functions, how come it was removed?

File: src/test/functional/functional-tests.js [hunk 4]
```diff
@@ -14,632 +14,686 @@
...
+const _ = require('lodash');
+const fs = require('fs');
+const http = require('http');
+const https = require('https');
+const url = require('url');
+//const chai = require('chai');
 import { assert } from 'chai'
-import superagent from 'superagent'
+const superagent = require('superagent');
+var uuid = require("uuid");
```
Comment (harshavardhana) [inline src/test/functional/functional-tests.js]: is there a reason to move to this style of package imports? due to avoiding ES6 transpiling stage?

**Comment by poornas** [comment]
@harshavardhana, the "require" style was used to make the functional tests work in mint because the ES6 transpilation steps were skipped. I have reverted the change to use imports now that the functional tests will continue to be maintained in the sdk itself.

**Comment by harshavardhana** [comment]
> @harshavardhana, the "require" style was used to make the functional tests work in mint because the ES6 transpilation steps were skipped. I have reverted the change to use imports now that the functional tests will continue to be maintained in the sdk itself.

Great @poornas

File: src/test/functional/functional-tests.js [hunk 5]
```diff
@@ -25,621 +24,674 @@ import fs from 'fs'
 import http from 'http'
 import https from 'https'
 import url from 'url'
-import { assert } from 'chai'
 import superagent from 'superagent'
+import { assert } from 'chai'
+
+var uuid = require("uuid");
```
Comment (harshavardhana) [inline src/test/functional/functional-tests.js]: You can change this to import style.

**Pull request merged**